### PR TITLE
q4 series

### PR DESCRIPTION
**Pull request by wave49192**

### CODE REVIEW

File: series.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+def series(nums, digit = 3)
+  nums = nums.to_a
+  nums.each_cons(digit) do |arr|
+    puts arr.join("")
+  end
+end
+
+series([1, 2, 3, 4, 5, 6], 5)
```
Comment (WatcharitNew) [inline series.rb:8]: Can you include result here as well? Like add it after this function in comment

File: series.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+def series(nums, digit = 3)
+  raise ArgumentError, "Number of digits must be greater than 0" unless digit > 0
+  nums = nums.to_s
+  raise ArgumentError, "Digit is greater than the length of the number" if digit > nums.length
+
+  result = []
+  nums.each_char.each_cons(digit) do |arr|
```
Comment (kan-datawow) [inline series.rb:7]: nice catch using `each_crons` we can shorter function by using map like this

```ruby
  num.chars.each_cons(length).map(&:join) 
```

File: series.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+def series(nums, digit = 3)
+  raise ArgumentError, "Number of digits must be greater than 0" unless digit > 0
+  nums = nums.to_s
```
Comment (WatcharitNew) [inline series.rb:3]: Please add empty error after raising error

Comment (WatcharitNew) [reply series.rb:3]: for `unless digit > 0`, IMO should change to `if digit <= 0` for more readable

File: spec/series_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require './series.rb'
+
+RSpec.describe "series" do
+  it 'should return array of correct values' do
+    expect(series(123456, 6)).to eq(["123456"])
```
Comment (WatcharitNew) [inline spec/series_spec.rb:5]: actually if have available time, may add more case  like (123456, 3) like this

**Review by WatcharitNew — APPROVED** [review]

File: spec/series_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  it 'should raise error if digit is greater than the length of the number' do
+    expect { series(123456, 7) }.to raise_error(ArgumentError, "Digit is greater than the length of the number")
+  end
+
+  it 'should raise error if digit less than or equal to 0' do
+    expect { series(123456, 0) }.to raise_error(ArgumentError, "Number of digits must be greater than 0")
+    expect { series(2342, -3) }.to raise_error(ArgumentError, "Number of digits must be greater than 0")
+  end
+end
```
Comment (numin123) [inline spec/series_spec.rb:17]:
```ruby
RSpec.describe "#series" do
  context "valid digit" do
    it 'returns full number' do
      expect(series(123456, 6)).to eq(["123456"])
    end

    it 'returns subseries' do
      expect(series(2342, 3)).to eq(["234", "342"])
    end

    it 'handles multiple groups' do
      expect(series(987654321, 4)).to eq(["9876", "8765", "7654", "6543", "5432", "4321"])
    end
  end

  context "invalid digit" do
    it 'errors if too large' do
      expect { series(123456, 7) }.to raise_error(ArgumentError, "Digit is greater than the length of the number")
    end

    it 'errors if non-positive' do
      expect { series(123456, 0) }.to raise_error(ArgumentError, "Number of digits must be greater than 0")
      expect { series(2342, -3) }.to raise_error(ArgumentError, "Number of digits must be greater than 0")
    end
  end
end

```

Maybe you can refactor your test case format to make it easier to understand

**Pull request merged**